Automattic/wp-calypso
# Stats: Update MobilePromoCard component to use redirects

#### Proposed Changes

Removes the raw URLs in favour of the jetpack.com redirects. This allows us to update the URLs server-side without having to make code changes. Applies to QR codes as well.

#### Testing Instructions



* Check out this branch.
* Run `yarn workspace @automattic/components run storybook`.
* Visit the Mobile Promo Card entry.
* Confirm the updated QR codes display correctly.
* Confirm the QR codes link out correctly. (they go via jetpack.com/redirect now)
* Use the Chrome dev tools and change to the "iPhone 12 Pro" setting. Refresh the view and confirm you get an App Store badge that links to the app page.
* Change to the "Pixel 5" setting and refresh again. Confirm you get a Play Store badge and that it's wired up correctly.
* For a12s, visit jetpack.com/redirect to confirm redirects are working.


#### Pre-merge Checklist





Replaces redirect work in #70521. Blocker for #70463.
